Source/Core/DolphinQt/Achievements/AchievementProgressWidget.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+#include "DolphinQt/Achievements/AchievementProgressWidget.h"
+
+#include <QCheckBox>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QProgressBar>
+#include <QPushButton>
+#include <QString>
+#include <QVBoxLayout>
+
+#include <fmt/format.h>
+
+#include <rcheevos/include/rc_api_runtime.h>
+#include <rcheevos/include/rc_api_user.h>
+#include <rcheevos/include/rc_runtime.h>
+
+#include "Core/AchievementManager.h"
+#include "Core/Config/AchievementSettings.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/Core.h"
+
+#include "DolphinQt/Config/ControllerInterface/ControllerInterfaceWindow.h"
+#include "DolphinQt/QtUtils/ModalMessageBox.h"
+#include "DolphinQt/QtUtils/NonDefaultQPushButton.h"
+#include "DolphinQt/QtUtils/SignalBlocking.h"
+#include "DolphinQt/Settings.h"
+
+AchievementProgressWidget::AchievementProgressWidget(QWidget* parent) : QWidget(parent)
+{
+  m_common_box = new QGroupBox();
+  m_common_layout = new QVBoxLayout();
+
+  UpdateData();
+
+  m_common_box->setLayout(m_common_layout);
+
+  auto* layout = new QVBoxLayout;
+  layout->setContentsMargins(0, 0, 0, 0);
+  layout->setAlignment(Qt::AlignTop);
+  layout->addWidget(m_common_box);
+  setLayout(layout);
+}
+
+QGroupBox*
+AchievementProgressWidget::CreateAchievementBox(const rc_api_achievement_definition_t* achievement)
+{
+  QLabel* a_title =
+      new QLabel(QString::fromLocal8Bit(achievement->title, strlen(achievement->title)));
+  QLabel* a_description = new QLabel(
+      QString::fromLocal8Bit(achievement->description, strlen(achievement->description)));
```

Comment:
What encoding is returned from your API? Is it not UTF8?

Reply:
Please never use `fromLocal8Bit`. Even if the API isn't using UTF-8, using a function that's dependent on what the current code page is is always a bad idea.

Reply:
Not sure how I reached the conclusion to use fromLocal8Bit but I'll change it to fromUTF8.